### ISSUE TITLE
`jax.linear_transpose` fails on map2alm

### ISSUE THREAD

**Issue by sbiquard** [CMBSciPol/jax-healpy#17]
This code fails on current main, but succeeds on 189d50a; thus it must be linked to the handling of masked pixels, 24d530a? @ArtemBasyrov 

```python
import jax
import jax.numpy as jnp
import jax_healpy as jhp

jax.config.update('jax_enable_x64', True)

NSIDE = 16
LMAX = 2 * NSIDE - 1
NPIX = 12 * NSIDE**2

x = jax.random.normal(jax.random.key(0), (NPIX,))

def f(m):
    return jhp.map2alm(m, lmax=LMAX, pol=False)

transpose = jax.linear_transpose(f, x)
transpose(f(x))
```

**Comment by ArtemBasyrov** [comment]
Fixed in [PR 20](https://github.com/CMBSciPol/jax-healpy/pull/20)
